packages/allure-hermione/src/index.ts

```diff
@@ -0,0 +1,318 @@
+/* eslint no-underscore-dangle: 0 */
+/* eslint import/order: 0 */
+import * as os from "node:os";
+import * as process from "node:process";
+import Hermione from "hermione";
+import {
+  AllureResults,
+  AllureRuntime,
+  AllureTest,
+  Attachment,
+  AttachmentMetadata,
+  LabelName,
+  LinkType,
+  ParameterOptions,
+  Stage,
+  Status,
+  TestResult,
+} from "allure-js-commons";
+import { ALLURE_METADATA_CONTENT_TYPE } from "allure-js-commons/internal";
+
+export interface AllureInMemoryWriter {
+  results: TestResult[];
+  attachments: Attachment[];
+  writeResult: (result: AllureResults) => void;
+  writeAttachment: (name: string, content: string, type: string) => void;
+}
+
+export interface HermioneAllureRuntime extends Omit<AllureRuntime, "writer"> {
+  writer: AllureInMemoryWriter;
+}
+
+export interface HermioneAllure extends Hermione {
+  allure: HermioneAllureRuntime;
+}
+
+export type HermioneAttachment = {
+  source: string;
+  mimetype: string;
+  encoding: BufferEncoding;
+};
+
+export type HermioneAttachmentMetadata = AttachmentMetadata & {
+  attachment?: HermioneAttachment[];
+};
+
+export type HermioneAttachmentMessage = {
+  testId: string;
+  metadata: HermioneAttachmentMetadata;
+};
+
+export type AllureReportOptions = {
+  resultsDir?: string;
+};
+
+export const getSuitePath = (test: Hermione.Test): string[] => {
+  const path = [];
+  let currentSuite = test.parent as Hermione.Suite;
+
+  while (currentSuite) {
+    if (currentSuite.title) {
+      path.unshift(currentSuite.title);
+    }
+
+    currentSuite = currentSuite.parent as Hermione.Suite;
+  }
+
+  return path;
+};
+
+const runningTests: Map<string, AllureTest> = new Map();
+
+const hermioneAllureReporter = (hermione: HermioneAllure, opts: AllureReportOptions) => {
```

Comment:
All other reporters are implemented as classes. They feels very sraightforward for me. Maybe we can do the same way here?
Its not a big deal and and PR can be merged without it.

Reply:
I can implement the report as class, but the function binding is necessary here.